client/src/pages/WeeklyTable.js

```diff
@@ -1,22 +1,53 @@
 import React from "react";
 
-import TableHeading from "./TableHeading";
+import "./WeeklyTable.css";
+
+const week = [
+	"2022-01-17T00:00:00.000Z",
+	"2022-01-18T00:00:00.000Z",
+	"2022-01-19T00:00:00.000Z",
+	"2022-01-20T00:00:00.000Z",
+	"2022-01-21T00:00:00.000Z",
+];
+
+const getBookingsByRow = (bookings, rowsCount, week) => {
+	const bookingsByDay = {};
+	week.forEach((day) => {
+		bookingsByDay[day] = bookings.filter((booking) => booking.date === day);
+	});
+
+	const bookingsByRow = [];
+	for (let i = 0; i < rowsCount; i++) {
+		bookingsByRow[i] = week.map((day) => bookingsByDay[day][i]);
+	}
+	return bookingsByRow;
+};
+
+const WeeklyTable = ({ bookings, rowsCount }) => {
```

Comment:
I don't think here is the right place to put rowsCount -> rowsCount is most easily obtained once you have a bookingsByRow object